Sources/TreeSitterClient/TreeSitterClient.swift

```diff
@@ -117,11 +117,7 @@ extension TreeSitterClient {
                                  limit: Int,
                                  readHandler: @escaping Parser.ReadBlock,
                                  completionHandler: @escaping () -> Void) {
-        guard let oldEndPoint = oldEndPoint else {
-            assertionFailure("oldEndPoint unavailable")
-            return
-        }
-
+        let oldEndPoint = oldEndPoint ?? .zero
         self.oldEndPoint = nil
```

Comment:
There's now some slightly confusing variable shadowing going on here. Any ideas on how to improve that?

Reply:
Added self on the property to make it more clear